app/grandchallenge/cases/models.py

```diff
@@ -1106,29 +1148,47 @@ def handle_failed_job(self, *, event):
     def delete_image_sets(*, job_summary):
         from grandchallenge.cases.tasks import delete_healthimaging_image_set
 
-        image_sets = job_summary.get("imageSetsSummary", [])
-        for image_set in image_sets:
+        for image_set_summary in job_summary.get("imageSetsSummary", []):
             on_commit(
                 delete_healthimaging_image_set.signature(
-                    kwargs=dict(image_set_id=image_set["imageSetId"])
+                    kwargs=dict(image_set_id=image_set_summary["imageSetId"])
                 ).apply_async
             )
 
     @staticmethod
-    def revert_image_set_to_initial_version(*, image_set):
+    def revert_image_set_to_initial_version(*, image_set_summary):
         from grandchallenge.cases.tasks import (
             revert_image_set_to_initial_version,
         )
 
         on_commit(
             revert_image_set_to_initial_version.signature(
                 kwargs=dict(
-                    image_set_id=image_set["imageSetId"],
-                    version_id=image_set["imageSetVersion"],
+                    image_set_id=image_set_summary["imageSetId"],
+                    version_id=image_set_summary["imageSetVersion"],
                 )
             ).apply_async
         )
 
-    @staticmethod
-    def convert_image_set_to_internal(*, image_set):
-        pass
+    def convert_image_set_to_internal(self, *, image_set_id):
+        dicom_image_set = DICOMImageSet.objects.create(
+            image_set_id=image_set_id,
+            dicom_image_set_upload=self,
+        )
+
+        Image.objects.create(
+            dicom_image_set=dicom_image_set,
+            # name="",  # todo: get name for set during upload
+            color_space=ColorSpace.GRAY.value,
+            width=-1,
+            height=-1,
+        )
```

Comment:
The new objects should be validated (make sure the caller of `convert_image_set_to_internal` can handle validation errors okay). This would need to be performed atomically, but I think we already are here? We shouldn't be forced to set any other values, `None` should be the value for things not filled out.

```suggestion
        dicom_image_set = DICOMImageSet(
            image_set_id=image_set_id,
            dicom_image_set_upload=self,
        )
        dicom_image_set.full_clean()
        dicom_image_set.save()

        image = Image(
            dicom_image_set=dicom_image_set,
            # name="",  # todo: get name for set during upload
        )
        image.full_clean()
        image.save()
```

Reply:
Thanks! 

> make sure the caller of convert_image_set_to_internal can handle validation errors okay

Currently handling this by catching everything in `handle_event` and marking the upload as failed if an error occurs. Should I add explicit handling of validation errors?